jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/IteratingCallbackTest.java

```diff
@@ -45,6 +49,37 @@ public void dispose() throws Exception
         scheduler.stop();
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testIterateWhileProcessingLoopCount(boolean succeededWinsRace)
+    {
+        var icb = new IteratingCallback()
+        {
+            int counter = 0;
+
+            @Override
+            protected Action process()
+            {
+                int counter = this.counter++;
+                if (counter == 0)
+                {
+                    iterate();
+                    if (succeededWinsRace)
+                        succeeded();
+                    else
+                        scheduler.schedule(this::succeeded, 100, TimeUnit.MILLISECONDS);
```

Comment:
I think it's too short to guarantee that succeeded() loses the race, I'd bump to 500.

Reply:
You're right, and relying on time is wrong when we can actually wait for a state change. Let me change that.